crates/rpc-types/src/eth/transaction/receipt.rs

```diff
@@ -1,63 +1,65 @@
-use crate::{other::OtherFields, Log};
-use alloy_primitives::{Address, Bloom, B256, U128, U256, U64, U8};
+use crate::other::OtherFields;
+use alloy_consensus::ReceiptEnvelope;
+use alloy_primitives::{Address, B256};
 use serde::{Deserialize, Serialize};
 
 /// Transaction receipt
-#[derive(Clone, Default, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct TransactionReceipt {
+    /// The receipt envelope, which contains the consensus receipt data..
+    #[serde(flatten)]
+    pub inner: ReceiptEnvelope,
+
     /// Transaction Hash.
     pub transaction_hash: B256,
     /// Index within the block.
-    pub transaction_index: U64,
+    #[serde(with = "alloy_serde::u64_hex_or_decimal")]
```

Comment:
should enforce QUANTITY on rpc
```suggestion
    #[serde(with = "alloy_serde::u64_hex")]
```